bin/isolated-functions.sh

```diff
@@ -339,15 +339,8 @@ ebegin() {
 	[[ ${RC_ENDCOL} == "yes" ]] && echo >&2
 	LAST_E_LEN=$(( 3 + ${#RC_INDENTATION} + ${#msg} ))
 	LAST_E_CMD="ebegin"
-	if [[ -v EBEGIN_EEND ]] ; then
-		# Already a call to ebegin
-		local prev="${EBEGIN_EEND[-1]}"
-		if [[ "${prev}" == "${FUNCNAME[1]}" ]] ; then
-			eqawarn "QA Notice: ebegin called in ${prev}, but missing call to eend (${FUNCNAME[1]})"
-		fi
-		EBEGIN_EEND+=( "${FUNCNAME[1]}" )
-	else
-		EBEGIN_EEND=( "${FUNCNAME[1]}" )
+	if (( __EBEGIN_EEND_COUNT++ > 0 )); then
+		eqawarn "QA Notice: nested ebegin call in ${FUNCNAME[1]}"
```

Comment:
In #824 I recall being told on #gentoo-qa that properly nested ebegin-eend pairs should be allowed (it's definitely used in some places). But you also [commented](https://github.com/gentoo/portage/pull/824#issuecomment-1105638609) that maybe it would be better to ban them and fix eclasses that do this.

This change would definitely start generating QA warnings, but I don't know that there's consensus to ban nested ebegin-eend.

Reply:
I opened #846 to simply revert your changes.

Reply:
I also updated this PR to only emit the nesting warning when ebegin is called twice in the same function.

Reply:
Alright.